src/confluent_kafka/admin/_acl.py

```diff
@@ -0,0 +1,268 @@
+    DESCRIBE = ACL_OPERATION_DESCRIBE  # : DESCRIBE operation
+    CLUSTER_ACTION = ACL_OPERATION_CLUSTER_ACTION  # : CLUSTER_ACTION operation
+    DESCRIBE_CONFIGS = ACL_OPERATION_DESCRIBE_CONFIGS  # : DESCRIBE_CONFIGS operation
+    ALTER_CONFIGS = ACL_OPERATION_ALTER_CONFIGS  # : ALTER_CONFIGS  operation
+    IDEMPOTENT_WRITE = ACL_OPERATION_IDEMPOTENT_WRITE  # : IDEMPOTENT_WRITE operation
+
+    def __lt__(self, other):
+        if self.__class__ != other.__class__:
+            return NotImplemented
+        return self.value < other.value
+
+
+class AclPermissionType(Enum):
+    """
+    Enumerates the different types of ACL permission types.
+    """
+    UNKNOWN = ACL_PERMISSION_TYPE_UNKNOWN  # : Unknown
+    ANY = ACL_PERMISSION_TYPE_ANY  # : In a filter, matches any AclPermissionType
+    DENY = ACL_PERMISSION_TYPE_DENY  # : Disallows access
+    ALLOW = ACL_PERMISSION_TYPE_ALLOW  # : Grants access
+
+    def __lt__(self, other):
+        if self.__class__ != other.__class__:
+            return NotImplemented
+        return self.value < other.value
+
+
+@functools.total_ordering
+class AclBinding(object):
+    """
+    Represents an acl binding that specify the operation and permission type for a specific principal
+    over one or more resources of the same type. Used by create_acls, returned by describe_acls and delete_acls.
+
+    Parameters
+    ----------
+    restype : `ResourceType`
+        The resource type.
+    name : `str`
+        The resource name, which depends on the resource type. For RESOURCE_BROKER, the resource name is the broker id.
+    resource_pattern_type : `ResourcePatternType`
+        The resource pattern, relative to the name.
+    principal : `str`
+        The principal this AclBinding refers to.
+    host : `str`
+        The host that the call is allowed to come from.
+    operation: `AclOperation`
+        The operation/s specified by this binding.
+    permission_type: `AclPermissionType`
+        The permission type for the specified operation.
+    """
+
+    def __init__(self, restype, name,
+                 resource_pattern_type, principal, host,
+                 operation, permission_type):
+        """
+        :param ResourceType restype: Resource type.
+        :param str name: The resource name, which depends on restype.
+                         For RESOURCE_BROKER, the resource name is the broker id.
+        :param ResourcePatternType resource_pattern_type: The resource pattern, relative to the name.
+        :param str principal: The principal this AclBinding refers to.
+        :param str host: The host that the call is allowed to come from.
+        :param AclOperation operation: The operation/s specified by this binding.
+        :param AclPermissionType The permission type for the specified operation.
+        """
+        super(AclBinding, self).__init__()
+
+        self._set_attrs(restype, name, resource_pattern_type, principal, host,
```

Comment:
It was to avoid repeating code because the validation checks are different, but I've found another way by adding the methods _not_none_args, _string_args and _forbidden_enums and overriding in the subclass